docs/guides/how_to_guides/configuring_data_contexts/how_to_instantiate_a_data_context_without_a_yml_file.rst

```diff
@@ -3,7 +3,193 @@
+
+        from great_expectations.data_context.types.base import DataContextConfig
+        from great_expectations.data_context import BaseDataContext
+
+        data_context_config = DataContextConfig(
+            datasources={
+                "my_sqlalchemy_datasource": DatasourceConfig(
+                    class_name="SqlAlchemyDatasource",
+                    credentials={
+                        "drivername": "custom_drivername",
+                        "host": "custom_host",
+                        "port": "custom_port",
+                        "username": "${USERNAME_FROM_ENVIRONMENT_VARIABLE}",
+                        "password": "${PASSWORD_FROM_ENVIRONMENT_VARIABLE}",
+                        "database": "custom_database",
+                    },
+                )
+            },
+            store_backend_defaults=S3StoreBackendDefaults(default_bucket_name="my_default_bucket"),
+        )
+
+    The following example shows a Data Context configuration with a pandas datasource and local filesystem defaults for metadata stores. Note: imports are omitted in the following examples.
+
+    .. code-block:: python
+
+        data_context_config = DataContextConfig(
+            datasources={
+                "my_pandas_datasource": DatasourceConfig(
+                    class_name="PandasDatasource",
+                    batch_kwargs_generators={
+                        "subdir_reader": {
+                            "class_name": "SubdirReaderBatchKwargsGenerator",
+                            "base_directory": "../data/",
+                        }
+                    },
+                )
+            },
+            store_backend_defaults=FilesystemStoreBackendDefaults(),
+    )
+
+    The following example shows a Data Context configuration with an sqlalchemy datasource and two GCS buckets for metadata stores, using some custom and some default prefixes. Note that you can still substitute environment variables as in the YAML based configuration to keep sensitive credentials out of your code. ``default_bucket_name``, ``default_project_name`` sets the default value for all stores that are not specified individually.
+    The resulting DataContextConfig from the following example creates an expectation store and data docs using the ``my_default_bucket`` and ``my_default_project`` parameters since their bucket and project is not specified explicitly. The validations store is created using the explicitly specified ``my_validations_bucket`` and ``my_validations_project``. Further, the prefixes are set for the expectations store and validations store, while data docs use the default ``data_docs`` prefix.
+
+    .. code-block:: python
+
+        data_context_config = DataContextConfig(
+            datasources={
+                "my_sqlalchemy_datasource": DatasourceConfig(
+                    class_name="SqlAlchemyDatasource",
+                    credentials={
+                        "drivername": "custom_drivername",
```

Comment:
Just a nitpick, would be nice to standardize how "user input" is identified in the examples. E.g. use my_x or REPLACE ME consistently, or add the "TODO: replace..." everywhere, or... pick your favorite. I know we do it differently in different docs, but I think it would be good to be consistent across this one page.